app/scripts/filters/resources.js

```diff
@@ -475,6 +475,15 @@ angular.module('openshiftConsole')
                 Navigate.projectOverviewURL(''); // fail case will invoke default route (projects list)
     };
   })
+  .filter('catalogURL', function(Navigate) {
+    return function(project) {
+      return angular.isString(project) ?
+        Navigate.catalogURL(project) :
+        angular.isObject(project) ?
+          Navigate.catalogURL(project.metadata && project.metadata.name):
+          Navigate.catalogURL(''); // fail case will invoke default route (projects list)
```

Comment:
> fail case will invoke default route (projects list)

Looks like that's not the case looking at the implementation below.

Reply:
I realize this was copied from the overview method, but I think something like this is better:

```js
if (!_.isString(project)) {
  project = _.get(project, 'metadata.name');
}
return Navigate.catalogURL(project);
```

Or possibly move that logic into `Navigate.catalogURL`

Reply:
I've moved it into Navigate.catalogURL